lib/ohai/plugins/linux/network.rb

```diff
@@ -406,17 +412,25 @@ def parse_ip_addr_inet6_line(cint, iface, line)
             ]
           end.first
 
-          unless route.nil? or route.empty?
+          if route && !route.empty?
             if family[:name] == "inet"
               ipaddress route[:src]
-              macaddress iface[route[:dev]][:addresses].select{|k,v| v["family"]=="lladdr"}.first.first unless iface[route[:dev]][:flags].include? "NOARP"
+              m = get_mac_for_interface(iface, route[:dev])
+              Ohai::Log.debug("Overwriting macaddress #{macaddress} with #{m} from interface #{route[:dev]}") if macaddress
+              macaddress m
             else
               ip6address route[:src]
+              if macaddress
+                Ohai::Log.debug("Not setting macaddress from ipv6 interface #{route[:dev]} because macaddress is already set")
+              else
+                macaddress get_mac_for_interface(iface, route[:dev])
+              end
             end
+          else
+            Ohai::Log.debug("Unable to determine ideal default route, not setting ipaddress/ip6address/macaddress. Should have considered #{default_route[:dev]}?")
```

Comment:
Allow me to prove the point I laid out in the original bug:

```
# pwd
/opt/chef/embedded/apps/ohai/lib/ohai/plugins/linux
# diff -u network.rb.orig network.rb
--- network.rb.orig	2015-12-23 15:00:11.551300264 -0800
+++ network.rb	2015-12-23 14:59:39.864068269 -0800
@@ -332,6 +332,8 @@
             else
               ip6address route[:src]
             end
+          else
+            puts "PHILD: I am here"
           end
         end
       end
# ohai | grep PHILD
PHILD: I am here
# 
```

I assure I while I said my patch wasn't good enough for production, I didn't just throw unreachable code in....